packages/tooling/fast-tooling/src/message-system/message-system.ts

```diff
@@ -79,23 +85,54 @@ export default class MessageSystem<C = {}> {
     }
 
     /**
-     * Post a message
+     * Post a message to the message system web worker
      */
     public postMessage(message: MessageSystemIncoming): void {
         if ((window as any).Worker && this.worker) {
-            this.worker.postMessage(message);
+            const uuid: string = uniqueId();
+
+            this.messageQueue[1].push(uuid);
+            this.worker.postMessage([message, uuid]);
         }
     }
 
     /**
-     * The onmessage handler for the message system
+     * The onmessage handler for the message system which recieves a message
+     * from the message system web worker and passes it to each registered item
      */
     private onMessage = (e: MessageEvent): void => {
-        this.register.forEach((registeredItem: Register) => {
-            registeredItem.onMessage(e);
-        });
+        this.messageQueue[0][e.data[1]] = e;
+
+        this.sendNextMessage();
     };
 
+    /**
+     * Fire the messages in the order they have been received when they are made available
+     */
+    private sendNextMessage = (): void => {
+        if (this.messageQueue[1][0] && this.messageQueue[0][this.messageQueue[1][0]]) {
+            const updatedEvent = new MessageEvent("message", {
+                data: this.messageQueue[0][this.messageQueue[1][0]].data[0],
```

Comment:
You might consider storing `this.messageQueue[0][this.messageQueue[1][0]` as a const so you you don't need to index the array so many times